### PR TITLE
Fix of floating-point special values

### PR DESCRIPTION
**Pull request by Checkmate50**
As the title suggests, this pull request fixed some issues occurring with special floating-point values (e.g. NaN and +/-infty).

### CODE REVIEW

File: Source/Basetypes/BigFloat.cs [hunk 1]
```diff
@@ -123,10 +123,10 @@ public static BigFloat FromBigDec(BigDec v, int significandSize, int exponentSiz
        */
       BIM sig, exp;
       int sigSize, expSize;
-      bool isNeg;
+      bool isNeg;
```
Comment (zvonimir) [inline Source/Basetypes/BigFloat.cs]: You should remove these trailing spaces here.

File: Test/floats/float4.bpl [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
 	d := 0NaN53e11;
 	d := 0nan53e11;
+    //assert d != 0NaN53e11;
 	d := 0+oo53e11;
+    assert d == 0e2047f53e11;
 	d := 0-oo53e11;
+    assert d == -0e2047f53e11;
 	d := -200e2000f53e11;
-	
+
```
Comment (zvonimir) [inline Test/floats/float4.bpl]: Trailing spaces again.

File: Test/floats/float4.bpl [hunk 3]
```diff
@@ -1,19 +1,25 @@
 // RUN: %boogie -proverWarnings:1 "%s" > "%t"
 // RUN: %diff "%s.expect" "%t"
-procedure foo() returns (r : float8e24) {
+procedure foo() returns (r : float24e8) {
```
Comment (zvonimir) [inline Test/floats/float4.bpl]: Indentation of this file looks messy. Are you maybe using tabs??
Please clean this up.

File: Test/floats/float4.bpl [hunk 4]
```diff
@@ -1,19 +1,25 @@
...
+    assert r != 0NaN24e8;  //NaN != NaN
+	r := 0+oo24e8;
+    assert r == 0e255f24e8;
+	r := 0-oo24e8;
+    assert r == -0e255f24e8;
+	r := -5e255f24e8;
 	
 	d := 0NaN53e11;
 	d := 0nan53e11;
+    //assert d != 0NaN53e11;
```
Comment (zvonimir) [inline Test/floats/float4.bpl]: I guess you should probably remove this comment.

File: Source/VCExpr/SimplifyLikeLineariser.cs [hunk 5]
```diff
@@ -392,6 +392,7 @@ public class SimplifyLikeExprLineariser : IVCExprVisitor<bool, LineariserOptions
     internal const string floatGeqName = "floatGeq";
     internal const string floatGtName = "floatGt";
     internal const string floatEqName = "floatEq";
+    internal const string floatNeqName = "floatEq";
```
Comment (zvonimir) [inline Source/VCExpr/SimplifyLikeLineariser.cs]: This seems to be a copy-paste typo, no? You are using the same string (`floatEq`) in both cases. Is that expected?

Comment (Checkmate50) [inline Source/VCExpr/SimplifyLikeLineariser.cs]: Oh, good catch, you are absolutely right!

**Comment by zvonimir** [comment]
@delcypher Could we maybe merge this now? Thanks!

**Pull request merged**